### DISCUSSION TITLE
How to set headers for response in a page

### DISCUSSION THREAD

**Discussion by dbzx10299** [nuxt/nuxt#25425] [Questions] [answered]
I have a site that will be statically generated and I see that the headers can be read with `useRequestHeaders`, but is there a way to also set a header on the page. I can't use the server directory since the site will be on netlify with `npm run generate` command.

I used this code and I did manage to get the headers to show up in the network tab but this feels a bit too verbose, is there a better way to do this?

```js
export default defineNuxtPlugin(nuxtApp => {
  if (process.server) {
    console.log(nuxtApp.ssrContext.event.node.res.setHeader('Accept-CH', 'Viewport-Width'))
  }
})
```

**Comment by ttonyh** [comment]
I would love it if we could set headers, like this, in Nuxt Config.

**Comment by dbzx10299** [comment, marked as answer]
You can now with route rules

```js
export default defineNuxtConfig({
  routeRules: {
    '/**': {
      headers: {
        'Accept-CH': 'Width, Viewport-Width'
      }
    }
  }
})
```
